src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -370,13 +380,22 @@ public function denormalize(mixed $data, string $type, string $format = null, ar
         $object = $this->instantiateObject($normalizedData, $type, $context, $reflectionClass, $allowedAttributes, $format);
         $resolvedClass = $this->objectClassResolver ? ($this->objectClassResolver)($object) : \get_class($object);
 
+        $flattenNestedAttributes = $context[self::FLATTEN_NESTED_ATTRIBUTES] ?? $this->defaultContext[self::FLATTEN_NESTED_ATTRIBUTES] ?? false;
+        if ($flattenNestedAttributes) {
+            $nestedAttributes = $this->getNestedAttributes($resolvedClass);
+        }
+
         foreach ($normalizedData as $attribute => $value) {
             $attributeContext = $this->getAttributeDenormalizationContext($resolvedClass, $attribute, $context);
 
             if ($this->nameConverter) {
                 $attribute = $this->nameConverter->denormalize($attribute, $resolvedClass, $format, $attributeContext);
             }
 
+            if ($flattenNestedAttributes && \in_array(strtolower($attribute), $nestedAttributes)) {
```

Comment:
This can be simplified with my proposals:

```suggestion
            if (isset($nestedAttributes[strtolower($attribute)])) {
```